ImageFeed/Services/ProfileImageService.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+final class ProfileImageService {
+    static let profileService = ProfileService.shared
+    private let storage: OAuth2TokenStorage = OAuth2TokenStorage()
+    static let shared = ProfileImageService()
+    static let DidChangeNotification = Notification.Name(rawValue: "ProfileImageProviderDidChange")
+    private var _avatarURL: String?
+    var avatarURL: String? {
+        get {
+            return _avatarURL
+        }
+        set {
+            _avatarURL = newValue
+        }
+    }
+
+    func fetchProfileImageURL(username: String, completion: @escaping (Result<UserResult,Error>) -> Void) {
+        if let token = storage.token {
+            let PublicProfileURL = URL(string: "https://api.unsplash.com/users/\(username)")
+            let req: RequestFactoryProtocol = RequestFactory()
+
+            guard let request = req.createRequest(
+                url: PublicProfileURL!,
```

Comment:
если тебе в username придет строка примерно такого вида {!!!!} – получишь за анврап падение в рантайме

давай писать безопасный код. 

Каждый анврап в тестовом проекте, проекте из портфолио - должен быть уместным, ты должен уметь объяснить почему это здесь можно было сделать. Джун, который бездумно будет форс анврапить на проекте – никому не нужен. 

Литература, как раз по твоему примеру
https://www.hackingwithswift.com/quick-start/understanding-swift/when-should-you-force-unwrap-optionals-in-swift


```suggestion
            let PublicProfileURL = URL(string: "https://api.unsplash.com/users/\(username)")
            let req: RequestFactoryProtocol = RequestFactory()

            guard let request = req.createRequest(
                url: PublicProfileURL!,
```

Reply:
PublicProfileURL - в верхнем camelCase, тоже надо поправить, это константа, а не тип, нечего ей с большой буквы называться